### ISSUE TITLE
mAP of the trained model

### ISSUE THREAD

**Issue by hustzxd** [1991viet/Yolo-pytorch#6]
Hi, thanks for your wonderful work！
In general, `YOLO` and `SSD` are trained based on a backbone network. So, we fisrt train the classerfier (Darknet-19 or VGG), then train the detector.  And in this project, we can `train model from scratch`,  does it mean initializing parameters randomly? Besides, the results of single picture and video are really good. Is there a 'mAP' result for the trained models?
Best wishes.

**Comment by 1991viet** [comment]
Hi,
Thanks for your question. 
If you want, of course we could train Darknet or VGG or Resnet, but pytorch already provides such pre-trained models (https://pytorch.org/docs/stable/torchvision/models.html), then we could conveniently use them.
For the question of training model from scratch: Yes. though it did make me surprised, you can train model without using any pre-trained model (even without initializing weight with normal distribution or bias with zero or sth like that). It will take a bit longer time to converge, but it works :) Maybe you could try one, and enjoy the output :)
Unfortunately, I haven't implement the part of mAP calculation. Currently I am busy with other 2 projects. I will back to this model after finishing them.
Bests,

**Comment by hustzxd** [comment]
Thanks for your reply. This project is very helpful for me.

**Comment by 1991viet** [comment]
I will email you after I update the mAP evaluation.
Bests,

**Comment by xjsxujingsong** [comment]
hi both, 
I tried to export the mAP calculation from https://github.com/amdegroot/ssd.pytorch. 
Using your pre-trained model on VOC2007 only, the mAP is only 0.6111.
I am trying to retrain it with 07+12 dataset. I use your pre-trained model and fine tune it. Now, after 70 epoch, the mAP is 64.2. Very similar to YOLOv1. A large gap to v2.

**Comment by 1991viet** [comment]
Hi,
Thank you for the information. One possible reason is that, I haven't implemented the part of changing input's size (in the paper, it is said that different input's sizes of factor of 32 are used). Additionally, I havent fine tuned the **conf_threshold** parameter yet (though I admit that it can not influence mAP too much). Finally, the learning rate schedule, I believe it could be improved as well. I will adapt these modification when having time.
Bests,

**Comment by pengfeidip** [comment]
hi , I come again ..
In the original paper, the YOLO v2 (darknet) is trained on ImageNet as a classifier, then trained on VOC and COCO as a dector(multiresolution).... so for the author, the **scratch** is really **scratch**.. I think the model you have trained is from **scratch** too, so what's  your **scratch** ?

**Comment by 1991viet** [comment]
Hi,
I think you agree that training from scratch means you train your model from the beginning, without using (loading) any pre-trained weights, right ?
In the original paper, the way YOLOv2 trained, you stated correctly.
In this implementation, you have 2 option when training your own model: using pre-trained weights I provided, or do not use any pre-trained weights (and this case is my **scratch**).

**Comment by pengfeidip** [comment]
ok I see..

![image](https://user-images.githubusercontent.com/23066797/49433016-f97e9d80-f7eb-11e8-996f-2f33b9e5d11f.png)
for example, model in ` voc2007` folder means the model  is only trained **only on** VOC 2007... so this's one of reasons  why there is a gap to the result in paper..

**Comment by 1991viet** [comment]
Yes, the name of folder indicates which dataset used. I didnt train with VOC2007+12. If you trained, please update result here. Thanks :)

**Comment by pengfeidip** [comment]
ok, Thank you~
